adapter-base-quarkus/src/main/java/org/eclipse/hono/adapter/quarkus/AbstractProtocolAdapterApplication.java

```diff
@@ -472,4 +484,49 @@ private ClientConfigProperties commandResponseSenderConfig() {
         return props;
     }
 
+    /**
+     * Creates resource limit checks based on data retrieved from a Prometheus server
+     * via its HTTP API.
+     *
+     * @param config The configuration properties.
+     * @param tenantClient The client to use for retrieving tenant configuration data.
+     * @throws NullPointerException if any of the parameters are {@code null}-
+     * @return The checks.
+     */
+    protected ResourceLimitChecks prometheusResourceLimitChecks(
+            final PrometheusBasedResourceLimitChecksConfig config,
+            final TenantClient tenantClient) {
+
+        Objects.requireNonNull(config);
+        Objects.requireNonNull(tenantClient);
+
+        if (config.isHostConfigured()) {
+            final WebClientOptions webClientOptions = new WebClientOptions();
+            webClientOptions.setConnectTimeout(config.getConnectTimeout());
+            webClientOptions.setDefaultHost(config.getHost());
+            webClientOptions.setDefaultPort(config.getPort());
+            webClientOptions.setTrustOptions(config.getTrustOptions());
+            webClientOptions.setKeyCertOptions(config.getKeyCertOptions());
+            webClientOptions.setSsl(config.isTlsEnabled());
+
+            final var webClient = WebClient.create(vertx, webClientOptions);
+            final var cacheTimeout = Duration.ofSeconds(config.getCacheTimeout());
+            final Caffeine<Object, Object> builder = Caffeine.newBuilder()
+                    // make sure we run one Prometheus query at a time
+                    .executor(Executors.newSingleThreadExecutor())
```

Comment:
You might prefer a daemon thread pool. If you have an orderly server shutdown, you might also register it explicitly.

Reply:
Not sure what you mean by
> you might also register it explicitly

What would need to be done for that?

Reply:
In my work code base, we use Guava’s ServiceManager to start/stop the application. An orderly server shutdown could call the executor’s shutdown to stop processing new requests. This isn’t necessary, just an FYI.

The non daemon thread could cause an orderly shutdown to hang as it won’t die until killed. So you either change the type or shutdown cleanly.

Reply:
Ok got it. I had already changed the *ThreadFactory* to return daemon threads so this should work now.